x-pack/plugins/security/server/routes/deprecations/kibana_user_role.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  SecurityGetRoleMappingResponse,
+  SecurityGetUserResponse,
+} from '@elastic/elasticsearch/api/types';
+
+import type { RouteDefinitionParams } from '..';
+import { KIBANA_ADMIN_ROLE_NAME, KIBANA_USER_ROLE_NAME } from '../../deprecations';
+import { getDetailedErrorMessage, wrapIntoCustomErrorResponse } from '../../errors';
+import { createLicensedRouteHandler } from '../licensed_route_handler';
+
+/**
+ * Defines routes required to handle `kibana_user` deprecation.
+ */
+export function defineKibanaUserRoleDeprecationRoutes({ router, logger }: RouteDefinitionParams) {
+  router.post(
+    {
+      path: '/internal/security/deprecations/kibana_user_role/_fix_users',
+      validate: false,
+    },
+    createLicensedRouteHandler(async (context, request, response) => {
+      let users: SecurityGetUserResponse;
+      try {
+        users = (await context.core.elasticsearch.client.asCurrentUser.security.getUser()).body;
+      } catch (err) {
+        logger.error(`Failed to retrieve users: ${getDetailedErrorMessage(err)}.`);
+        return response.customError(wrapIntoCustomErrorResponse(err));
+      }
+
+      const usersWithKibanaUserRole = Object.values(users).filter((user) =>
+        user.roles.includes(KIBANA_USER_ROLE_NAME)
+      );
+
+      if (usersWithKibanaUserRole.length === 0) {
+        logger.debug(`No users with "${KIBANA_USER_ROLE_NAME}" role found.`);
+      } else {
+        logger.debug(
```

Comment:
__question:__ are we concerned with mentioning user names in the logs in this case?